packages/bson-transpilers/test/yaml/driver-syntax.yaml

```diff
@@ -72,6 +72,23 @@ tests:
                       'x': new NumberLong(0), 
                       'y': new ObjectId()
                     });"
+                php: |-
+                    // Requires the MongoDB PHP Driver
+                    // https://www.mongodb.com/docs/drivers/php/
+
+                    $client = new \MongoDB\Client('mongodb://localhost:27017');
+                    $db = $client->selectDatabase('db');
+                    
+                    $convertPipelineArray = function ($aggregationData) {
+                        return array_map(function ($key, $value) {
+                                return [$key => $value];
+                            },
+                            array_keys($aggregationData),
+                            $aggregationData
+                        );
+                    };
+
+                    $result = $db->selectCollection('collection')->aggregate($convertPipelineArray(['x' => 0, 'y' => new \MongoDB\BSON\ObjectId()]));
```

Comment:
Unfortunately Aggregate helper has it's own pipeline check stage and it falls with error because of pipeline keys. This $convertPipelineArray function resolves this problem and convert array:
````
[
    'x' => 0,
    'y' => new ObjectId()
]
````
into
````
[
    ['x' => 0],
    ['y' => new ObjectId()]
]
````
which is ok for Aggregate pipeline checker.

Reply:
If we assume `x` and `y` above are placeholders for pipeline stages (e.g. `$match`, `$project`), I think there is a problem well before we even get to invoking `$convertPipelineArray`.

A valid pipeline can contain multiple instances of the same stage, which is why it is defined as an array of objects instead of simply an object. If Compass is generating an associative array with pipeline stages occupying the same namespace (top-level array keys), then a valid pipeline could be mangled (i.e. stages lost) before we get a chance to call `$convertPipelineArray` or otherwise generate the `aggregate()` method call.

Reply:
Once again... According to what we have in the test input (``aggregation: '{x: NumberLong(0), y: ObjectId()}'``), there is an object. According to convert rules in PHP it becomes to:
````
(object) ['x' => 0, 'y' => new ObjectId()]
````
Aggregate helper definitely can't work with object, so I transformed it into array:
````
['x' => 0, 'y' => new ObjectId()]
````
But with this type of array Aggregate helper still can't work, because of existing keys, which don't satisfy pipeline keys requirements. That's why the transformation function appeared. How do you think, what should we do here?

Reply:
>  To be safe from potential errors the pipeline is covered with check for array form

I assume you're referring to the `toPHPArray` helper function in `PHPUtils.js`, which is called on `spec.aggregation` in `DriverTemplate` (near the top of `packages/bson-transpilers/symbols/php/templates.yaml`. I'm still not sure why that is necessary, as I'd expect Compass to always provide a valid pipeline array. This may be a question for the Compass reviewer, so I'll ask in the diff where `toPHPArray(spec.aggregation)` is called.

OK to resolve this thread.